### PR TITLE
JavaScript capstone project - API-based webapp

### PR DESCRIPTION
**Pull request by MarcoDiaz2000**
Project created by: @Bifabig  and @MarcoDiaz2000 

The final JavaScript project is to create our own web application based on an external API. We selected an API that provides data about Star Wars movies and built the web application around it. The web application has 2 user interfaces:

- A home page showing a list of items that you can "like."
- A popup window with more data about an item that you can use to comment on it or reserve it for a period of time.

### Project requirements

**APIs**

First, you need to find an API so you can base the development of the webapp around it. The API should allow you to:

Get a list of items with a unique item id (or generate the unique id).
For a given item, get detailed information about it.
We recommend that you choose an API that doesn't need authentication. if you select an API that requires authentication, you should implement it on your own. Also, if you select an API that provides image resources your webapp will be more visual.

**You should build these interfaces:**
- The home page.
- The comments popup.

You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

**Home page**
When the page loads, the webapp retrieves data from:
- The selected API and shows the list of items on screen.
- The Involvement API to show the item likes.
Remember that your page should make only 2 requests:
- One to the base API.
- And one to the Involvement API.
When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
When the user clicks on the "Comments" button, the Comments popup appears.
Home page header and navigation similar to the given mockup.
Home page footer similar to the given mockup.

**Comments popup**
When the popup loads, the webapp retrieves data from:
- The selected API and shows details about the selected item.
- The Involvement API to show the item comments.
When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

**Counters We have counters in all the interfaces that show:**
- The number of items (home).
- The number of comments (comments popup).
Even if the API gives you these numbers, you will create a specific function to calculate these numbers in each page. These count functions need to be covered with unit tests using Jest.

### CODE REVIEW

File: src/modules/card.js
Comment (KDlamini) [inline src/modules/card.js:1]: Well done on your design!!! 💯 

Kindly add spaces next to each card so that they do not overlap with each other.

image

Design:

image

File: src/modules/commentCounter.test.js
Comment (KDlamini) [inline src/modules/commentCounter.test.js:1]: Great job with covering unit tests using Jest! 👏🏾 

Kindly make sure that you have tests for all edge cases. This means that you should have at least a scenario in which there are 0 items on the page (the counter function should return 0).

File: src/modules/spaceships.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import spaceship1 from '../images/spaceship1.png';
+import spaceship2 from '../images/spaceship2.png';
+import spaceship3 from '../images/spaceship4.png';
+
+function createSpaceships() {
```
Comment (KDlamini) [inline src/modules/spaceships.js]: Kindly use ES6 syntax for your application.  This means using `arrow functions` instead of the `function` keyword. 

Your function should be `const createSpaceships = () => {...};`

- Kindly apply that to all your files including the `showNotification`, `mobileMenu`, `insertLogo` functions etc.

File: src/modules/apiTvmaze.js
Comment (KDlamini) [inline src/modules/apiTvmaze.js:1]: Kindly use ES6 syntax to fetch API data. This means that you should use arrow functions and `async/await` to fetch data.

Your function should look similar to this:

```JavaScript
const fetchMovies = async() => {
  const response = await fetch('https://api.tvmaze.com/search/shows?q=star%20wars')
    
  return await response.json();
}
```

File: src/index.js
Comment (KDlamini) [inline src/index.js:1]: Same here, kindly use arrow functions and `async/await` to GET or POST your data.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @MarcoDiaz2000, @Bifabig  🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working web app with Webpack and API integration. 🎊 
There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Good PR title and description
✔️  Good API integration
✔️  Functions for creating comments and content counters work as expected
✔️  User can like and comment successfully. Well done adding functionality for limiting likes 🔥
✔️  Home page header and footer are present 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @MarcoDiaz2000 and @KDlamini  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Change made as requested by the previous reviewer :+1: 
- Unit tests are passing :heavy_check_mark: 
- Project is working as expected :surfer: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
